text/0497-glue-l2-construct.md

```diff
@@ -0,0 +1,418 @@
+provide extra jars, and a description
+
+```
+glue.Job(this, 'pySparkFlexEtlJob', {
+   jobType: glue.JobType.FLEX,
+   jobLanguage: glue.JobLanguage.SCALA_SPARK,
+   glueVersion: glue.GlueVersion.V3_0,
+   scriptS3Url: 's3://bucket-name/path-to-python-script',
+   extraJarsS3Url: ['s3://bucket-name/path-to-extra-python-scripts'],
+   description: 'an example pySpark ETL job',
+   numberOfWorkers: 20,
+});
+
+new glue.Job(this, 'FlexJob', {
+   jobType: glue.JobType.FLEX,
+   jobLanguage: glue.JobLanguage.PYSPARK,
+   glueVersion: glue.GlueVersion.V3_0,
+   pythonVersion: glue.PythonVersion.3_6,
+   scriptS3Url: 's3://bucket-name/path-to-python-script',
+   description: 'an example Flex job',
+   numberOfWorkers: 20,
+});
+```
+
+### Python Shell Jobs
+
+A Python shell job runs Python scripts as a shell and supports a Python
+version that depends on the AWS Glue version you are using. This can be used
+to schedule and run tasks that don't require an Apache Spark environment.
+Python 3.6 and 3.9 are supported.
+
+We’ll default to `PythonVersion.3_9` which customer can override. Python
+shell jobs doesn’t support different worker type, instead it has MaxDPU
+feature. Customer can choose Max DPU = `0.0625` or Max DPU = `1`. By default
+MaxDPU will be set `0.0625`. Also `PythonVersion.3_9` supports preloaded
+analytics libraries using flag `library-set=analytics` , this feature will
+be enable by default.
+
+```
+new glue.Job(this, 'PythonShellJob', {
+    jobType: glue.JobType.PYTHON_SHELL,
+    jobLanguage: glue.JobLanguage.PYSPARK,
+    scriptS3Url: 's3://bucket-name/path-to-python-script',
+});
+```
+
+Optional overrides:
+
+```
+new glue.Job(this, 'PythonShellJob', {
+    jobType: glue.JobType.PYTHON_SHELL,
+    jobLanguage: glue.JobLanguage.PYSPARK,
+    glueVersion: glue.GlueVersion.V1_0,
+    pythonVersion: glue.PythonVersion.3_6,
+    scriptS3Url: 's3://bucket-name/path-to-python-script',
+    description: 'an example Python Shell job',
+    numberOfWorkers: 20,
+});
+```
+
+### Ray Jobs
+
+Glue ray only supports Z.2X worker type and 4.0 Glue version. Runtime
+will default to `Ray2.3` and min workers will default to 3.
+
+```
+declare const bucket: s3.Bucket;
+new glue.Job(this, 'GlueRayJob', {
+  jobType: glue.JobType.GLUE_RAY,
+  jobLanguage: glue.JobLanguage.PYTHON,
+  scriptS3Url: 's3://bucket-name/path-to-python-script',
+});
+```
+
+Optionally customer can override min workers and other Glue job fields
+
+```
+declare const bucket: s3.Bucket;
+new glue.Job(this, 'GlueRayJob', {
+  jobType: glue.JobType.GLUE_RAY,
+  jobLanguage: glue.JobLanguage.PYTHON,
+  runtime: glue.Runtime.RAY_2_2
+  scriptS3Url: 's3://bucket-name/path-to-python-script',
+  minWorkers: 20,
+  numberOfWorkers: 50
+});
+```
+
+### Job Triggers
+
+We will add convenience functions for adding triggers to jobs. Standalone
+triggers are an anti-pattern, so we will only create triggers from within a
+workflow.
```

Comment:
As mentioned above, I'm not seeing a section on `Workflows` so I think I'm missing a little context here, however, instead of creating functions for each type of trigger, I'd suggest making `trigger`\`triggers` an optional prop on jobs that takes in an ITrigger and then do something more along the lines of:
```
trigger: Trigger.onDemand()
```
or 
```
trigger: Trigger.scheduled(...)
```
or 
```
trigger: Trigger.conditional(...)
```
In addition, you could also have a convenience function like:
```
myGlueJob.addTrigger(Trigger.notifyEvent(...));
```

Reply:
Adding function at job level will be ambiguous, we'll have to add additional parameter to distinguish between action/predicate list remove vs add for removing existing jobs from trigger or add new jobs to list. Rather I think we should add trigger functions in workflow